crates/ty_python_semantic/resources/mdtest/descriptor_protocol.md

```diff
@@ -170,6 +170,8 @@ def f1(flag: bool):
             attr = DataDescriptor()
 
         def f(self):
+            # TODO: Here we treat attr as if it's a data descriptor. But it should fallback to instance attribute.
+            # error: [invalid-assignment] "Invalid assignment to data descriptor attribute `attr` on type `Self@f` with custom `__set__` method"
```

Comment:
I think the new diagnostic is fine, because *if* `flag` is `True`, then `self.attr` is a `DataDescriptor`, and in that case, we would call its `__set__` method with an invalid type.

```suggestion
            # error: [invalid-assignment] "Invalid assignment to data descriptor attribute `attr` on type `Self@f` with custom `__set__` method"
```